modal/image.py

```diff
@@ -843,7 +853,9 @@ def add_local_python_source(
         ```
         """
         mount = _Mount._from_local_python_packages(*modules, ignore=ignore)
-        return self._add_mount_layer_or_copy(mount, copy=copy)
+        img = self._add_mount_layer_or_copy(mount, copy=copy)
+        img._added_python_source_set |= set(modules)
```

Comment:
This is the key part of this change - track all `add_local_python_source` modules on the Image objects so we can inspect it later when the automount logic executes to see which modules are missing. It's not perfect, but people can get rid of false positive warnings by setting `include_source` to any boolean value (with the expectation that they know what they are doing)